### ISSUE TITLE
Explorer not displaying the recursive relationship correctly

### ISSUE THREAD

**Issue by semihsalihoglu-uw** [kuzudb/explorer#158]
I'm running this query both in KuzuExplorer (latest, see screenshot) and CLI (0.4.2). Explorer does not seem to draw the output recursive rel as expected. Here's the query:

```
MATCH (p1:Person)-[o1:Owns]->(a1:Account)-[r:Transfer* SHORTEST]-(a2:Account)<-[o2:Owns]-(p2:Person) 
WHERE p1.email = "georodaw366@hotmail.com" AND p2.email = "ezimmerman@yahoo.com" 
RETURN *;
```
The output is as follows on the shell and contains 3 internal edges that bind to `r`:
<img width="1596" alt="Screen Shot 2024-07-04 at 4 33 11 PM" src="https://github.com/kuzudb/explorer/assets/26987866/5a2d1fbc-4556-4769-b0a9-e874436d1b2a">

However the output on Explorer shows only 1 edge (the middle edge). The other two edges are not drawn:
<img width="1739" alt="Screen Shot 2024-07-04 at 4 34 10 PM" src="https://github.com/kuzudb/explorer/assets/26987866/9d363b40-e295-446f-a214-0e47937e850d">

The table view of Explorer seems to have the correct output with 3 edges:

<img width="1744" alt="Screen Shot 2024-07-04 at 4 35 43 PM" src="https://github.com/kuzudb/explorer/assets/26987866/f0417cec-0092-4339-a38a-1235efb0a782">

You can replicate by cloning the datasets and graph here: https://github.com/kuzudb/cypher-workshop/tree/main (except I made 2 changes to the datasets: 1) Removed the last transfer in transfer.csv "5,8,257,47231669"; and 2) changed the address of "+10,Rhea,"1919 Joshua Square Suite 643, Carmen Town, WY 83111",WY,83111,rh2schultz@gmail.com" so that Rhea lives in the same address as George (georodaw366@hotmail.com). This creates a second 5-length path between George and Edward (ezimmerman@yahoo.com).

<img width="1157" alt="Screen Shot 2024-07-04 at 4 29 50 PM" src="https://github.com/kuzudb/explorer/assets/26987866/a366b407-ab16-433d-a064-51d7faa822e3">

**Comment by mewim** [comment]
There are two issues. 
1. The kernel seems to return edges in both directions, which @andyfengHKU will address.
2. Explorer is handling multiple edges between nodes incorrectly. It should not draw these edges with overlaps. I am working on fixing it.

Also, it might be nice to add a feature to trace/highlight each path returned from recursive query. I will create a new issue for this.

**Comment by mewim** [comment]
Overlapping edge handling has been addressed in #164.
<img width="2672" alt="Screenshot 2024-07-05 at 11 22 57 AM" src="https://github.com/kuzudb/explorer/assets/14037726/54f2279f-bc39-4147-9067-69b0d47e7431">

**Comment by mewim** [comment]
This is fixed with the latest nightly build `0.4.3-dev.47` and `dev` version of Explorer:
<img width="1624" alt="Screenshot 2024-07-06 at 9 13 33 PM" src="https://github.com/kuzudb/explorer/assets/14037726/66245abe-dbda-4ed8-96eb-7dd5f182aebf">
